paimon-core/src/main/java/org/apache/paimon/schema/SchemaValidation.java

```diff
@@ -92,6 +92,15 @@ public static void validateTableSchema(TableSchema schema) {
                             WRITE_MODE.key(), APPEND_ONLY, CHANGELOG_PRODUCER.key()));
         }
 
+        if (options.writeMode() == WriteMode.AUTO
+                && CollectionUtil.isNullOrEmpty(schema.primaryKeys())
+                && changelogProducer != ChangelogProducer.NONE) {
+            throw new UnsupportedOperationException(
+                    String.format(
+                            "Can not set %s on table without primary keys.",
```

Comment:
We could improve the exception message to help users to solve, like `when set write mode to none, the table must have primary key` etc. Meanwhile, adds this information into document to reduce the situation that the users have this validation failure.